networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/subnets/ValidatorBasedStableSubnetSubscriber.java

```diff
@@ -120,8 +145,43 @@ private Set<SubnetSubscription> adjustNumberOfSubscriptionsToNumberOfValidators(
     return newSubnetSubscriptions;
   }
 
+  private UInt256 extractNodeId() {
+    final ECPoint ecPoint =
+        Functions.publicKeyToPoint(
+            discoveryNodeId.orElseThrow(
+                () -> new IllegalArgumentException("Unable to get discovery node id")));
+    final Bytes pubKeyUncompressed = Bytes.wrap(ecPoint.getEncoded(false)).slice(1);
+    BigInteger nodeId = new BigInteger(pubKeyUncompressed.toArray());
+    return UInt256.valueOf(nodeId);
+  }
+
+  private List<UInt64> computeSubscribedSubnets(final UInt256 nodeId, final UInt64 epoch) {
```

Comment:
I'd move these and other calculation helpers to something like MiscHelpers, so we could test it carefully

Reply:
Moved to `MiscHelpers`